logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MessageSendFailureHandler.kt

```diff
@@ -54,4 +75,30 @@ class MessageSendFailureHandlerImpl internal constructor(
                         clientRepository.storeUserClientIdList(entry.key, entry.value)
                     }
             }
+
+    override suspend fun handleFailureUpdateMessageStatus(
+        failure: CoreFailure,
+        conversationId: ConversationId,
+        messageId: String,
+        messageType: String
+    ) {
+        when (failure) {
+            is NetworkFailure.FederatedBackendFailure -> {
+                kaliumLogger.i("Sending message of type $messageType failed due to federation context availability.")
+                messageRepository.updateMessageStatus(MessageEntity.Status.FAILED_REMOTELY, conversationId, messageId)
+            }
+            else -> {
+                messageRepository.updateMessageStatus(MessageEntity.Status.FAILED, conversationId, messageId)
+            }
+        }
+        if (failure is CoreFailure.Unknown) {
+            kaliumLogger.e(
+                "There was an unknown error trying to send the message of type: $messageType, cause: $failure",
+                failure.rootCause
+            )
+            failure.rootCause?.printStackTrace()
```

Comment:
why are we printing the error stack trace twice?

Reply:
Right, I just moved the code without thinking much about it, but I think we don't want that, so I removed it 👍